pkg/resource/resource.go

```diff
@@ -0,0 +1,118 @@
+package resource
+
+import (
+	"github.com/IBM-Cloud/bluemix-go/api/resource/resourcev1/catalog"
+	"github.com/IBM-Cloud/bluemix-go/api/resource/resourcev1/controller"
+	"github.com/IBM-Cloud/bluemix-go/api/resource/resourcev1/management"
+	"github.com/IBM-Cloud/bluemix-go/models"
+	"github.com/IBM-Cloud/bluemix-go/session"
+	"k8s.io/klog/v2"
+	"os"
+)
+
+func CreateServiceInstance(c *session.Session, Name, ServiceName, ServicePlan, ResourceGrp, Region, apikey string) (string, error) {
+	catalogClient, err := catalog.New(c)
+
+	if err != nil {
+		return "", err
+	}
+	resCatalogAPI := catalogClient.ResourceCatalog()
+
+	service, err := resCatalogAPI.FindByName(ServiceName, true)
+	if err != nil {
+		return "", err
+	}
+
+	servicePlanID, err := resCatalogAPI.GetServicePlanID(service[0], ServicePlan)
+	if err != nil {
+		return "", err
+	}
+	if servicePlanID == "" {
+		_, err := resCatalogAPI.GetServicePlan(ServicePlan)
```

Comment:
what the difference between GetServicePlan vs GetServicePlanID methods? do we need to have both methods? can we reduce to one function call for getting the ID?

Reply:
I have two calls since I have no idea when GetServicePlanID() returns empty string, if incase it return empty string the GetServicePlan function call will be used. For creation of s3 instance GetServicePlanID() function is enpugh to create.

Reply:
can we ask someone who owns the code when it returns empty and why?